lib/post_creator.rb

```diff
@@ -186,10 +186,8 @@ def create
         @post.link_post_uploads
         update_uploads_secure_status
         ensure_in_allowed_users if guardian.is_staff?
-        unarchive_message
-        if !@opts[:import_mode]
-          DraftSequence.next!(@user, draft_key)
-        end
+        unarchive_message if !@opts[:import_mode]
```

Comment:
Is this fixing a bug? Might be worth extracting into another (and much smaller) PR.

Reply:
This was a problem during the initial synchronization of a mailbox which would create a lot of posts and would sometime reset the archived status of a topic.